src/black/__init__.py

```diff
@@ -385,8 +386,29 @@ def main(
     if config and verbose:
         out(f"Using configuration from {config}.", bold=False, fg="blue")
     if code is not None:
-        print(format_str(code, mode=mode))
+        then = datetime.utcnow()
+
+        try:
+            formatted = format_file_contents(code, fast=fast, mode=mode)
+        except NothingChanged:
+            formatted = code
+
+        if write_back is WriteBack.CHECK:
+            ctx.exit(0 if code == formatted else 1)
+        elif write_back in (WriteBack.DIFF, WriteBack.COLOR_DIFF):
+            now = datetime.utcnow()
+            src_name = f"STDIN\t{then} +0000"
+            dst_name = f"STDOUT\t{now} +0000"
+
+            diff_contents = output.diff(code, formatted, src_name, dst_name)
+            if write_back == WriteBack.COLOR_DIFF:
+                diff_contents = color_diff(diff_contents)
+            print(diff_contents)
```

Comment:
I don't know whether `echo` is better than `print` here (and we use both inconsistently in the codebase) but `echo` seems to be safer?

Reply:
That may be the case, but if we're going to be using a textiowrapper on stdout, we could just use the write method, as is done in the linked example.

Reply:
*facepalm* should've caught that, my bad

Reply:
No longer relevant. 6221b88.